### PR TITLE
Warm instances integration

### PR DESCRIPTION
**Pull request by sasurobert**
Warm instances integration

### CODE REVIEW

File: factory/blockProcessorCreator.go [hunk 1]
```diff
@@ -49,7 +55,7 @@ func (pcf *processComponentsFactory) newBlockProcessor(
 	pendingMiniBlocksHandler process.PendingMiniBlocksHandler,
 	txSimulatorProcessorArgs *txsimulator.ArgsTxSimulator,
 	arwenChangeLocker common.Locker,
-) (process.BlockProcessor, process.VirtualMachinesContainerFactory, error) {
+) (*blockProcessorAndVmFactories, error) {
```
Comment (iulianpascalau) [inline factory/blockProcessorCreator.go:58]: :+1: :+1: :+1:

File: factory/blockProcessorCreator.go [hunk 2]
```diff
@@ -38,6 +38,12 @@ import (
 	"github.com/ElrondNetwork/elrond-vm-common/parsers"
 )
 
+type blockProcessorAndVmFactories struct {
+	blockProcessor         process.BlockProcessor
+	vmFactoryForTxSimulate process.VirtualMachinesContainerFactory
+	vmFactoryForProcessing process.VirtualMachinesContainerFactory
+}
+
 func (pcf *processComponentsFactory) newBlockProcessor(
```
Comment (iulianpascalau) [inline factory/blockProcessorCreator.go:47]: all these methods newBlockProcessor, newShardBlockProcessor, newMetaBlockProcessor can have a suffix like`AndVmFactories`

File: go.mod [hunk 3]
```diff
@@ -9,10 +9,10 @@ require (
 	github.com/ElrondNetwork/concurrent-map v0.1.3
 	github.com/ElrondNetwork/covalent-indexer-go v1.0.6
 	github.com/ElrondNetwork/elastic-indexer-go v1.1.32
-	github.com/ElrondNetwork/elrond-go-core v1.1.7
+	github.com/ElrondNetwork/elrond-go-core v1.1.8-0.20220112192542-db36097f7385
```
Comment (iulianpascalau) [inline go.mod:12]: do not forget about proper releases and after that `go mod tidy`

File: integrationTests/multiShard/block/executingMiniblocks/executingMiniblocks_test.go [hunk 4]
```diff
@@ -50,7 +50,7 @@ func TestShouldProcessBlocksInMultiShardArchitecture(t *testing.T) {
 
 	defer func() {
 		for _, n := range nodes {
-			_ = n.Messenger.Close()
+			n.Close()
```
Comment (iulianpascalau) [inline integrationTests/multiShard/block/executingMiniblocks/executingMiniblocks_test.go:53]: :+1:

File: process/factory/shard/vmContainerFactory.go [hunk 5]
```diff
@@ -165,7 +165,19 @@ func (vmf *vmContainerFactory) Create() (process.VirtualMachinesContainer, error
 
 // Close closes the vm container factory
 func (vmf *vmContainerFactory) Close() error {
-	return vmf.blockChainHookImpl.Close()
+	err := vmf.container.Close()
+	if err != nil {
+		logVMContainerFactory.Error("cannot close container", "error", err)
+		return err
```
Comment (iulianpascalau) [inline process/factory/shard/vmContainerFactory.go:171]: I would not return error here just to try to also close the blockchain hook.

**Review by iulianpascalau — COMMENTED** [review]
GJ :+1:

File: go.mod (hunk 3 above)
Comment (sasurobert) [reply go.mod]: done

File: process/factory/shard/vmContainerFactory.go (hunk 5 above)
Comment (sasurobert) [reply process/factory/shard/vmContainerFactory.go]: resolved

**Review by iulianpascalau — DISMISSED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by alyn509 — DISMISSED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by camilbancioiu — DISMISSED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by gabi-vuls — DISMISSED** [review]
System test passed

**Review by camilbancioiu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**